.github/actions/configure-snowflake-cli/action.yml

```diff
@@ -0,0 +1,110 @@
+name: "Configure Snowflake CLI"
+description: "Configures a ~/.snowflake/config.toml file and install the Snowflake CLI"
+inputs:
+  PRIVATE_KEY_PASSPHRASE:
+    description: "The pass phrase used to decrypt the private key"
+    required: true
+  PRIVATE_KEY:
+    description: "The private key"
+    required: true
+  ACCOUNT:
+    description: "Snowflake account identifier"
+    required: true
+  USER:
+    description: "Snowflake user identifier"
+    required: true
+  WAREHOUSE:
+    description: "Snowflake warehouse"
+    required: false
+    default: ""
+outputs:
+  config-file-path:
+    description: "The path to the Snowflake config.toml file"
+    value: ${{ steps.configure-snowflake.outputs.config-file-path }}
+  connection-file-path:
+    description: "The path to the Snowflake connections.toml file"
+    value: ${{ steps.configure-snowflake.outputs.connection-file-path }}
+  version:
+    description: A plain text description of the Snowflake CLI version
+    value: ${{ steps.connection-test.outputs.version }}
+  connection:
+    description: >
+      A pretty-printed description of the default Snowflake connection
+      encoded using base64. Decode using `base64 --decode`.
+    value: ${{ steps.connection-test.outputs.connection }}
+
+runs:
+  using: "composite"
+  steps:
+    - name: Create config.toml file
+      shell: bash
+      id: configure-snowflake
+      run: |
+        # Specify path for config.toml, connections.toml, and our private key
+        mkdir -p $HOME/.snowflake
+        CONFIG_FILE=$HOME/.snowflake/config.toml
+        CONNECTION_FILE=$HOME/.snowflake/connections.toml
+        PRIVATE_KEY_FILE=$HOME/.snowflake/rsa_key.p8
+      
+        # Write to connections.toml file
+        echo '[default]' >> $CONNECTION_FILE
+        echo "account = \"${{ inputs.ACCOUNT }}\"" >> $CONNECTION_FILE
+        echo "user = \"${{ inputs.USER }}\"" >> $CONNECTION_FILE
+        echo 'warehouse = "${{ inputs.WAREHOUSE }}"' >> $CONNECTION_FILE
```

Comment:
Syntax issue? `\"${{ inputs.WAREHOUSE }}\"`

Reply:
I'm not sure if we need to use double quotes throughout the whole expression to properly evaluate `${{ inputs.WAREHOUSE }}`, but it wouldn't hurt.